transmogrifier/sources/xml/datacite.py

```diff
@@ -339,6 +242,189 @@ def _get_contributors(cls, source_record: Tag) -> Iterator[timdex.Contributor]:
+                funder.funder_name = str(funder_name.string)
+            if award_number := funding_reference.find("awardNumber"):
+                funder.award_number = award_number.string or None
+                funder.award_uri = award_number.get("awardURI") or None
+            if funder_identifier := funding_reference.find(
+                "funderIdentifier", string=True
+            ):
+                funder.funder_identifier = str(funder_identifier.string)
+                funder.funder_identifier_type = (
+                    funder_identifier.get("funderIdentifierType") or None
+                )
+            if funder != timdex.Funder():
+                funding_information.append(funder)
+        return funding_information or None
+
+    @classmethod
+    def get_identifiers(
+        cls,
+        source_record: Tag,
+    ) -> list[timdex.Identifier] | None:
+        identifiers = []
+        if identifier_element := source_record.metadata.find("identifier", string=True):
+            identifiers.append(
+                timdex.Identifier(
+                    value=str(identifier_element.string),
+                    kind=identifier_element.get("identifierType") or "Not specified",
+                )
+            )
+        identifiers.extend(list(cls._get_alternate_identifiers(source_record)))
+        identifiers.extend(list(cls._get_related_identifiers(source_record)))
+        return identifiers or None
+
+    @classmethod
+    def _get_alternate_identifiers(
+        cls,
+        source_record: Tag,
+    ) -> Iterator[timdex.Identifier]:
+        for alternate_identifier_element in source_record.metadata.find_all(
+            "alternateIdentifier", string=True
+        ):
+            yield timdex.Identifier(
+                value=str(alternate_identifier_element.string),
+                kind=alternate_identifier_element.get("alternateIdentifierType")
+                or "Not specified",
+            )
+
+    @classmethod
+    def _get_related_identifiers(
+        cls,
+        source_record: Tag,
+    ) -> Iterator[timdex.Identifier]:
+        related_identifier_elements = source_record.metadata.find_all(
+            "relatedIdentifier", string=True
+        )
+        for related_identifier_element in [
+            related_identifier_element
+            for related_identifier_element in related_identifier_elements
+            if related_identifier_element.get("relationType") == "IsIdenticalTo"
+        ]:
+            yield timdex.Identifier(
+                value=cls.generate_related_item_identifier_url(
+                    related_identifier_element
+                ),
+                kind=str(related_identifier_element["relationType"]),
+            )
+
+    @classmethod
+    def get_languages(cls, source_record: Tag) -> list[str] | None:
+        languages = []
+        if language := source_record.metadata.find("language", string=True):
```

Comment:
[Non-blocking] When do we include the "_element" suffix to a variable?

* Some cases, we just use the field name:
   * [get_funding_information ](https://github.com/MITLibraries/transmogrifier/pull/181/files#diff-875ef9d9184fe0a891a9abf429a0b3187d70ef66c9908a9e0a4008cb26995afdR334)
   * get_languages
   * [get_edition](https://github.com/MITLibraries/transmogrifier/pull/181/files#diff-875ef9d9184fe0a891a9abf429a0b3187d70ef66c9908a9e0a4008cb26995afdR316)
* Other cases, we include the suffix "_element": 
   * [get_identifier](https://github.com/MITLibraries/transmogrifier/pull/181/files#diff-875ef9d9184fe0a891a9abf429a0b3187d70ef66c9908a9e0a4008cb26995afdR358) 
   * [_get_alternate_identifier](https://github.com/MITLibraries/transmogrifier/pull/181/files#diff-875ef9d9184fe0a891a9abf429a0b3187d70ef66c9908a9e0a4008cb26995afdR374-R376)

This is something I've personally been getting tripped up on in the transforms I've been working on --naming conventions assigned to `find`/`find_all` method calls. These differences don't cause any problems but consistency would be nice...

@ghukill @ehanson8

Reply:
I was using the `_element` convention where element attributes are accessed in addition to `element.string` so it probably should be used for `get_funding_information` as well. To me, it doesn't feel as necessary when we're just grabbing `element.string`. Interested in consistency as well so I want to hear everyone's thoughts

Reply:
I see the thought process there @ehanson8, and I agree that consistency would be nice, but it doesn't feel like a deal breaker to me with some variation.  Given how common the convention is to a) find an element, and then b) use that element variable to extract a string _or_ an attribute value, I think the `_element` could get pretty chatty.  

It might be worth an issue?  Then we could keep moving through this refactor, but make a global pass at some point.  

In summary: I'm personally okay with variations, given different times in codebae, but an issue to revist could be nice.

Reply:
Thanks for sharing! I'll apply that thought process as I work on other transforms, @ehanson8 .